tools/walletextension/userconn/user_conn.go

```diff
@@ -154,6 +166,15 @@ func (w *userConnWS) IsClosed() bool {
 	return w.isClosed
 }
 
+func (w *userConnWS) ReadRequestParams() map[string]string {
+	params := make(map[string]string)
+	queryParams := w.req.URL.Query()
+	for key, value := range queryParams {
+		params[key] = value[0]
+	}
+	return params
```

Comment:
these 2 functions seem identical.
They could both use a util function that takes a query and returns a map

Reply:
fixed.